webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/repository/ldap/LDAPRuleRepository.scala

```diff
@@ -169,6 +169,44 @@ class WoLDAPRuleRepository(
     }
   }
 
+  private[this] def internalRuleSystemDirective(id:RuleId, modId: ModificationId, actor:EventActor, reason:Option[String]) : IOResult[DeleteRuleDiff] = {
+    for {
+      con          <- ldap
+      entry        <- con.get(rudderDit.RULES.configRuleDN(id.value)).notOptional("rule with ID '%s' is not present".format(id.value))
+      oldCr        <- mapper.entry2Rule(entry).toIO.chainError("Error when transforming LDAP entry into a rule for id %s. Entry: %s".format(id, entry))
+      deleted      <- con.delete(rudderDit.RULES.configRuleDN(id.value)).chainError("Error when deleting system rule with ID %s".format(id))
+      diff         =  DeleteRuleDiff(oldCr)
+      loggedAction <- actionLogger.saveDeleteRule(modId, principal = actor, deleteDiff = diff, reason = reason)
+      autoArchive  <- ZIO.when(autoExportOnModify && deleted.nonEmpty  && !oldCr.isSystem) {
+        for {
+          commiter <- personIdentService.getPersonIdentOrDefault(actor.name)
+          archive  <- gitCrArchiver.deleteRule(id, Some((modId, commiter, reason)))
+        } yield {
+          archive
+        }
+      }
+    } yield {
+      diff
+    }
+  }
+
+  private[this] def deleteRuleCommon(ruleId: RuleId, modId: ModificationId, actor:EventActor, reason:Option[String], isSystem: Boolean) ={
```

Comment:
Should only call InternalDelete with boolean value to True to force system Rule Deletion

Reply:
and not be private